### PR TITLE
Release pipeline

### PR DESCRIPTION
**Pull request by lolski**
## What is the goal of this PR?

Release pipeline for `client-java`, which performs:
1. deployment of a Github draft release
2. Deployment of artifacts to Maven

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -83,11 +83,46 @@ jobs:
...
+  deploy-github:
+    machine: true
+    working_directory: ~/client-java
+    steps:
+      - install-bazel-linux-rbe
+      - checkout
+      - run: DEPLOY_GITHUB_TOKEN="$GRABL_CREDENTIAL" bazel run //:deploy-github
+
+  deploy-maven:
+    machine: true
```
Comment (haikalpribadi) [inline .circleci/config.yml]: where's the test equivalent of this job, @lolski ? I.e. `deploy-maven-test`? I see that it's still bundled up in `test-deployment-maven`. I think we should split `deploy-maven-test` and it should be as clean as this `deploy-maven`, except with argument `-- snapshot`

File: RELEASE_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Client-Java
+
+ { Please add the release notes here }
```
Comment (haikalpribadi) [inline RELEASE_TEMPLATE.md]: Are you planning to add the installation instruction here too? I think we should show how to declare the dependency (`<groupId></>`, `<artifactId></>`, `<version></>`) in their pom.xml (like how we show in docs)

File: WORKSPACE [hunk 3]
```diff
@@ -115,6 +115,30 @@ graknlabs_graql_maven_dependencies = "maven_dependencies")
...
+bazel_rules_docker_repositories = "repositories")
+bazel_rules_docker_repositories()
+
+load("@io_bazel_rules_docker//container:container.bzl", "container_pull")
+container_pull(
+  name = "openjdk_image",
+  registry = "index.docker.io",
+  repository = "library/openjdk",
+  tag = "8"
+)
```
Comment (haikalpribadi) [inline WORKSPACE]: same question as in https://github.com/graknlabs/graql/pull/28/files#r269271565

Why do we need these docker rules?

File: .circleci/config.yml (hunk 1 above)
Comment (lolski) [reply .circleci/config.yml]: good point

File: WORKSPACE (hunk 3 above)
Comment (lolski) [reply WORKSPACE]: fixed

File: RELEASE_TEMPLATE.md (hunk 2 above)
Comment (lolski) [reply RELEASE_TEMPLATE.md]: fixed

File: .circleci/config.yml (hunk 1 above)
Comment (lolski) [reply .circleci/config.yml]: fixed

**Pull request merged**